TextToSpeech/TextToSpeechJsonRpc.cpp

```diff
@@ -166,17 +169,56 @@ namespace Plugin {
     {
         if(_tts) {
         Exchange::ITextToSpeech::Configuration config;
-        config.ttsEndPoint = GET_STR(parameters, "ttsendpoint", "");
-        config.ttsEndPointSecured = GET_STR(parameters, "ttsendpointsecured", "");
+        std::string proxyTtsEndPoint = GET_STR(parameters, "ttsendpoint", "");
+        std::regex urlPattern(R"((https?|ftp):\/\/\S+\/\S*)",std::regex::icase);
+        if(regex_match(proxyTtsEndPoint,urlPattern)) {
+            config.ttsEndPoint = proxyTtsEndPoint;
+        }
+        else {
+            config.ttsEndPoint = "";
+        }
+        std::string proxyTtsEndPointSecured = GET_STR(parameters, "ttsendpointsecured", "");
+        if(regex_match(proxyTtsEndPointSecured,urlPattern)) {
+            config.ttsEndPointSecured = proxyTtsEndPointSecured;
+        }
+        else {
+            config.ttsEndPointSecured = "";
+        }
         config.language = GET_STR(parameters, "language", "");
-        config.voice = GET_STR(parameters, "voice", "");
-        config.speechRate = GET_STR(parameters, "speechrate", "");
-        config.volume = (uint8_t) std::stod(GET_STR(parameters, "volume", "0.0"));
+        std::set<std::string> languageSet = {"en-US","es-MX","fr-CA","en-GB","de-DE","it-IT"};
+        config.language = languageSet.find(config.language) != languageSet.end() ? config.language : "";
+        std::string proxyVoice = GET_STR(parameters, "voice", "");
+        if(proxyVoice.c_str()[0] == ' ' || Utils::isValidUnsignedInt(const_cast<char*>(proxyVoice.c_str()))) {
+            config.voice = "";
+        }
+        else {
+            config.voice = proxyVoice;
+        }
+        std::string proxySpeechRate = GET_STR(parameters, "speechrate", "");
+        if(proxySpeechRate == "slow" || proxySpeechRate == "medium" || proxySpeechRate == "fast" || proxySpeechRate == "faster" || proxySpeechRate == "fastest") {
+            config.speechRate = proxySpeechRate;
+        }
+        else {
+            config.speechRate = "";
+        }
+        uint8_t proxyVolume = 0;
+        try {
+            proxyVolume = (uint8_t) std::stod(GET_STR(parameters, "volume", "0.0"));
```

Comment:
We should also try to formalize the type of the 'volume' configuration. I don't remember why it is kept as a string (i.e passed as string from ttsclient & gets converted to integer here vice versa. But the 'rate' configuration is properly sent as int to & fro client.